src/main/java/fr/cnes/sonar/plugins/icode/check/ICodeSensor.java

```diff
@@ -112,16 +109,15 @@ public void execute(SensorContext sensorContext) {
             final String outputOption = "-o";
             final String command = String.join(" ", executable, String.join(" ",files), outputOption, outputPath);
 
-            LOGGER.info("running i-Code CNES and generating results to "+ outputPath);
-            LOGGER.debug("command : " + command);
+            LOGGER.info("Running i-Code CNES and generating results to "+ outputPath);
             try {
                 final Process icode =  Runtime.getRuntime().exec(command);
                 int success = icode.waitFor();
-                    if(success==0){
-                        LOGGER.info("Auto-launch successfully executed i-Code CNES");
-                    }else{
-                        LOGGER.error("i-Code CNES auto-launch analysis failed with exit code "+success);
-                    }
+                if(0!=success){
+                    final String message = String.format("i-Code CNES auto-launch analysis failed with exit code %d.",success);
+                    throw new RuntimeException(message);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define and throw a dedicated exception instead of using a generic one. ([squid:S00112](https://sonarcloud.io/organizations/lequal/rules#rule_key=squid%3AS00112))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fr.cnes.sonarqube.plugins%3Asonaricode&pullRequest=32&issues=AWXWzj9j_eEZPtUm_fKr&open=AWXWzj9j_eEZPtUm_fKr)

[ISSUE_KEY]: AWXWzj9j_eEZPtUm_fKr